### PR TITLE
add Redux Toolkit, RTK Query, Jotai, Valtio

### PR DESCRIPTION
**Pull request by phryneas**
## Type of change

- [x] Content addition
- [ ] Bug fix
- [ ] Behavior change

## Summary of change

This adds Redux Toolkit, RTK Query, Jotai and Valtio to the libraries.

## Checklist

- [x] I have added a line to the [changelog](../CHANGELOG.md) with the current
      date, change, PR number and author

- [x] The changes follow the [contributing guidelines](../CONTRIBUTING.md)
- [x] I have searched all existing content and verified my additions are novel
      and unique
- [ ] The content was added to the end of the appropriate data list (I added RTK & RTK Query next to Redux and Jotai/Valtio next to Zustand as all of these group together contextually)
- [x] All required content fields are accurately populated
- [x] All items are tagged with all relevant tags

### CODE REVIEW

File: packages/site/src/data/react/libraries.ts [hunk 1]
```diff
@@ -679,11 +701,31 @@ export const libraries: Library<typeof libraryTags[number]>[] = [
...
+	},
+	{
+		name: "Valtio",
+		author: "Poimandres",
+		gitHubRepo: "pmndrs/valtio",
+		npmPackage: "valtio",
+		href: "https://github.com/pmndrs/valtio",
+		image: "https://github.com/pmndrs.png",
+		tags: ["state management"],
+		description: "makes proxy-state simple",
```
Comment (jbachhardie) [inline packages/site/src/data/react/libraries.ts:727]: Could you maybe add a little bit more detail here? I feel like "proxy-state" is not a term widely understood enough to make the three-word description sufficient for most visitors.

**Review by jbachhardie — CHANGES_REQUESTED** [review]
Hi there and thank you so much for contributing. These are all great additions, there's just one thing that would be nice to see fleshed out a bit.

File: packages/site/src/data/react/libraries.ts (hunk 1 above)
Comment (phryneas) [reply packages/site/src/data/react/libraries.ts:727]: To be honest, I haven't used valtio myself - from what I understand, it it like the "Poimandres version of MobX", so another take might be "modern observables" or something like that, but I'm not really sure if that would be fitting. 

I just took the official headline from their homepage. Maybe @dai-shi could provide us with a more fitting description?

Comment (dai-shi) [reply packages/site/src/data/react/libraries.ts:727]: How about "Lightweight Proxy-Based State Management for ReactJS and VanillaJS"?

Comment (phryneas) [reply packages/site/src/data/react/libraries.ts]: sounds good, I've adjusted it accordingly

**Review by jbachhardie — APPROVED** [review]
Nice. Thank you for the changes.

**Pull request merged**